ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgentTest.java

```diff
@@ -321,6 +323,41 @@ public void incomingConnection_maxPeersExceeded()
     }
   }
 
+  @Test
+  public void incomingConnection_succeedsEventuallyWithRandomPeerPrioritization() {
+    // Saturate connections with one local and one remote
+    startAgentWithMaxPeers(2, builder -> builder.randomPeerPriority(true));
+    agent.connect(createPeer());
+    connectionInitializer.simulateIncomingConnection(connection(createPeer()));
+    // Sanity check
+    assertThat(agent.getConnectionCount()).isEqualTo(2);
+
+    boolean newConnectionDisconnected = false;
+    boolean oldConnectionDisconnected = false;
+    // With very high probability we should see the connections churn
+    for (int i = 0; i < 1000; ++i) {
```

Comment:
Perhaps we should white box this test.  

Go ahead and set up the pool and the peer randomly, but then get the peer that would be evicted (sorted last), then generate peers until we get one that we see is higher priority than the one to be evicted, and then _only once_ do the add check loop.  We would need to get at the peers list and the nodeIdMask to do this, hence the white boxing.

Reply:
We also need a test case where the new node doesn't get added because it's going to lose the comparison too.  So we need another test with the same white box setup but instead we search for a node that will rank below all other nodes.

Reply:
Is the only reason to make the test deterministic?

Reply:
If that's the only benefit, I think I prefer black boxing it because I've run into a lot of problems with changing code that has very whiteboxed testing in p2p. With the number of iterations we do here and the number of remote connections, this is an intermittent test with a very very low probability of failing.

Reply:
If we are going to black box it we still need to detect connection accepted and connection rejected.